st2common/st2common/models/db/execution_queue.py

```diff
@@ -0,0 +1,56 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+
+import mongoengine as me
+
+from st2common import log as logging
+from st2common.models.db import stormbase
+from st2common.models.db import MongoDBAccess
+from st2common.fields import ComplexDateTimeField
+from st2common.util import date as date_utils
+from st2common.constants.types import ResourceType
+
+__all__ = [
+    'ExecutionQueueDB',
+]
+
+
+LOG = logging.getLogger(__name__)
+
+
+class ExecutionQueueDB(stormbase.StormFoundationDB):
+    RESOURCE_TYPE = ResourceType.EXECUTION_REQUEST
+    UID_FIELDS = ['id']
+    liveaction = stormbase.EscapedDictField(required=True)
+    start_timestamp = ComplexDateTimeField(
+        default=date_utils.get_datetime_utc_now,
+        help_text='The timestamp when the liveaction was created.')
+    delay = me.IntField()
+    priority = me.IntField()
+    affinity = me.StringField()
```

Comment:
Priority will be (is, just needs API) working in this PR, but I'll remove affinity.

Reply:
Also, when I run a new action execution but I didn't specify a priority, what's the default priority?  Can I assume st2 will run the action execution immediately, which means the priority is high?  Also having priority means there is a finite capacity. At what point will st2 start putting action execution into the scheduling queue here? What if there's a constant stream of action execution with default priority but not enough capacity?  When will lower priority execution in the scheduling queue be run?

Reply:
Having said that, the current scope of the PR is "delay".  Delay is straightforward problem to solve. I'm hesitant adding "priority" into this PR here without a good understanding and solution for the problem.

Reply:
I'll leave it out to have the discussion in another PR, but these have pretty well-established answers. At least in situations I've worked in with the past. You just select your finite number of priorities and have a probability distribution from which priority to choose from on scheduling. I'm sure there are other ways, but I can't think of another that wouldn't require shared state. As said though I'll leave it out for now.

Reply:
Thanks! I think this will help progress this PR quicker. The priority problem is still yours. Just need to explain it a bit in more details to us what these well established answers are and how it will be implemented in st2.